terrarium/dl_gitmanager/dl_gitmanager/git_manager.py

```diff
@@ -114,23 +115,36 @@ def _iter_list_diffs(
             for _, diff_item in submodule_manager._iter_list_diffs(commits=sm_commits_for_all_commits):
                 yield submodule_base_path, diff_item
 
-    def _collect_paths_from_diffs(self, diff_iterable: Iterable[tuple[Path, Diff]]) -> list[str]:
+    def _collect_paths_from_diffs(
+        self, diff_iterable: Iterable[tuple[Path, Diff]], without_deleted: bool = False
+    ) -> list[str]:
         result: set[str] = set()
         for base_path, diff_item in diff_iterable:
             if diff_item.a_path:
                 result.add(str(base_path / diff_item.a_path))
             if diff_item.b_path:
                 result.add(str(base_path / diff_item.b_path))
 
+        if without_deleted:
+            result = {path for path in result if os.path.exists(path)}
```

Comment:
You need to check for existence via the git file tree for the `head` state because it will not necessarily be the same as the real FS (i.e. the current repository state might (and probably will) be different from `head`).